windows/src/desktop/setup/Keyman.Setup.System.SetupUILanguageManager.pas

```diff
@@ -103,6 +120,41 @@ class procedure TSetupUILanguageManager.RegisterSetupStrings(const tag: string;
   FLocales.Add(tag, locale[ssLanguageName]);
 end;
 
+class procedure TSetupUILanguageManager.Reorder;
+var
+  item: TPair<string,string>;
+  i: integer;
+  j: integer;
+  sortedKeys: TArray<string>;
+  sortedPairA: TList< TPair<string,string>>;
+begin
+  sortedPairA := TList< TPair<string,string>>.Create();
+  i := 0;
+  for item in TSetupUILanguageManager.Locales do
+  begin
+    sortedPairA.Add(item);
+  end;
+  sortedPairA.Sort(
+    TComparer<TPair<string, string>>.Construct(
+      function(const Left, Right: TPair<string, string>): Integer
+      begin
+        // Sort by the Key value
+        Result := CompareText(Left.Value, Right.Value);
+      end
+    )
+  );
+  FSortedValues := sortedPairA;
```

Comment:
`FSortedValues.Free` will need to be called at Destruction or some point to avoid Memory leaks.